### ISSUE TITLE
coredns crashes with api timeout

### ISSUE THREAD

**Issue by saravana1992** [kubernetes/kubeadm#1463]
Coredns crashes with the API time out and restarts frequently

Logs of coredns:

E0322 10:27:47.898277       1 reflector.go:205] github.com/coredns/coredns/plugin/kubernetes/controller.go:355: Failed to list *v1.Namespace: Get https://10.96.0.1:443/api/v1/namespaces?limit=500&resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
E0322 10:27:47.900119       1 reflector.go:205] github.com/coredns/coredns/plugin/kubernetes/controller.go:350: Failed to list *v1.Endpoints: Get https://10.96.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
E0322 10:27:47.900153       1 reflector.go:205] github.com/coredns/coredns/plugin/kubernetes/controller.go:348: Failed to list *v1.Service: Get https://10.96.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
E0322 10:28:18.899701       1 reflector.go:205] github.com/coredns/coredns/plugin/kubernetes/controller.go:355: Failed to list *v1.Namespace: Get https://10.96.0.1:443/api/v1/namespaces?limit=500&resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
E0322 10:28:18.902272       1 reflector.go:205] github.com/coredns/coredns/plugin/kubernetes/controller.go:350: Failed to list *v1.Endpoints: Get https://10.96.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
E0322 10:28:18.904249       1 reflector.go:205] github.com/coredns/coredns/plugin/kubernetes/controller.go:348: Failed to list *v1.Service: Get https://10.96.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
E0322 10:28:49.902399       1 reflector.go:205] github.com/coredns/coredns/plugin/kubernetes/controller.go:355: Failed to list *v1.Namespace: Get https://10.96.0.1:443/api/v1/namespaces?limit=500&resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
E0322 10:28:49.903769       1 reflector.go:205] github.com/coredns/coredns/plugin/kubernetes/controller.go:350: Failed to list *v1.Endpoints: Get https://10.96.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
E0322 10:28:49.906441       1 reflector.go:205] github.com/coredns/coredns/plugin/kubernetes/controller.go:348: Failed to list *v1.Service: Get https://10.96.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
2019/03/22 10:29:00 [INFO] SIGTERM: Shutting down servers then terminating

**vagrant@Master:~$ kubectl describe svc kubernetes**

Name:              kubernetes
Namespace:         default
Labels:            component=apiserver
                   provider=kubernetes
Annotations:       <none>
Selector:          <none>
Type:              ClusterIP
IP:                10.96.0.1
Port:              https  443/TCP
TargetPort:        6443/TCP
Endpoints:         192.168.33.51:6443
Session Affinity:  None
Events:            <none>

**Comment by neolit123** [comment]
please add details about your setup following our issue template.

**Comment by timothysc** [comment]
closing due to timeout.

**Comment by dolphyvn** [comment]
I having same issue when setup cluster with kubeadm (1.15.0 )
One cluster works, other not. Using same network configuration, same ansible playbook.
On the cluster where coredns crash due to can not connect to 10.96.0.1. Ican curl https://10.96.0.1 if I stand on the node or pods that run with hostnetwork mode ( like calico pods )

**Comment by cflev** [comment]
anyone found a solution for this? I have the same problem:
`2019-08-13T19:37:09.805Z [INFO] plugin/reload: Running configuration MD5 = 5d5369fbc12f985709b924e721217843
E0813 19:37:34.804922       1 reflector.go:134] github.com/coredns/coredns/plugin/kubernetes/controller.go:322: Failed to list *v1.Namespace: Get https://10.96.0.1:443/api/v1/namespaces?limit=500&resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
E0813 19:37:34.804922       1 reflector.go:134] github.com/coredns/coredns/plugin/kubernetes/controller.go:322: Failed to list *v1.Namespace: Get https://10.96.0.1:443/api/v1/namespaces?limit=500&resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout`

**Comment by dolphyvn** [comment]
1.15.0 seem to solved the problem with me.

**Comment by cflev** [comment]
Thanks for the reply,
It is a fresh install with 1.15.2

**Comment by dolphyvn** [comment]
@cflev actually you can try to route to that ip via your gateway. It works with me that way. But I dont want to do that so I update to 1.15.0 and it gone.

**Comment by cflev** [comment]
@dolphyvn Thanks for the tip!
I just checked the iptables and I saw some old entries, so I will revise them again :)

**Comment by 100cm** [comment]
@dolphyvn  how to route to that ip?

**Comment by dolphyvn** [comment]
@100cm just do a normal static route:
ip route add 10.96.0.1 via {gatewayip|device}

**Comment by 100cm** [comment]
```
cni0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 10.244.0.1  netmask 255.255.255.0  broadcast 0.0.0.0
        inet6 fe80::84ac:7cff:fe6f:4043  prefixlen 64  scopeid 0x20<link>
        ether 86:ac:7c:6f:40:43  txqueuelen 1000  (Ethernet)
        RX packets 8  bytes 224 (224.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 15  bytes 950 (950.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

docker0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255
        ether 02:42:e6:1d:0d:ed  txqueuelen 0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.0.180  netmask 255.255.255.0  broadcast 192.168.0.255
        inet6 fe80::f816:3eff:fe9a:8f4d  prefixlen 64  scopeid 0x20<link>
        ether fa:16:3e:9a:8f:4d  txqueuelen 1000  (Ethernet)
        RX packets 1342638  bytes 1742384482 (1.6 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 1502009  bytes 193154284 (184.2 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth0:1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 218.78.34.245  netmask 255.255.255.255  broadcast 218.78.34.245
        ether fa:16:3e:9a:8f:4d  txqueuelen 1000  (Ethernet)

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 74307  bytes 4674586 (4.4 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 74307  bytes 4674586 (4.4 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

ip route add 10.96.0.1 via 192.168.1.0
RTNETLINK answers: Network is unreachable

@dolphyvn  i'm so confused for this. how could i do next?

**Comment by Venkatsai** [comment]
Looks like this issue is not resolved in v1.17.3 kubeadm version. The AWS VPC CNI manages the pod IP assignment and the IP's are not overlapped. Could you let me know what logs can be shared to identify the core issue.
